.github/scripts/strategy-matrix/generate.py

```diff
@@ -238,21 +240,145 @@ def generate_strategy_matrix(all: bool, config: Config) -> list:
+                    {
+                        "config_name": config_name + "-tsan-ubsan",
+                        "cmake_args": configs["tsan_ubsan"],
+                        "cmake_target": cmake_target,
+                        "build_only": build_only,
+                        "build_type": build_type,
+                        "os": os,
+                        "architecture": architecture,
+                        "sanitizers": "Thread,UndefinedBehavior",
+                    }
+                )
+        else:
+            configurations.append(
+                {
+                    "config_name": config_name,
+                    "cmake_args": cmake_args,
+                    "cmake_target": cmake_target,
+                    "build_only": build_only,
+                    "build_type": build_type,
+                    "os": os,
+                    "architecture": architecture,
+                }
+            )
 
     return configurations
 
 
+def addSanitizerConfigs(
+    architecture: dict,
+    os: dict,
+    cmake_args: str,
+    cxx_flags: str,
+) -> dict:
+    extra_warning_flags = ""
+    linker_relocation_flags = ""
+    linker_flags = ""
+
+    # Use large code model to avoid relocation errors with large binaries
+    # Only for x86-64 (amd64) - ARM64 doesn't support -mcmodel=large
+    if architecture["platform"] == "linux/amd64" and os["compiler_name"] == "gcc":
+        # Add -mcmodel=large to both compiler AND linker flags
+        # This is needed because sanitizers create very large binaries and
+        # large model removes the 2GB limitation that medium model has
+        cxx_flags += " -mcmodel=large -fno-PIC"
+        linker_relocation_flags += " -mcmodel=large -fno-PIC"
+
+    # Create default sanitizer flags
+    sanitizers_flags = "undefined,float-divide-by-zero"
+
+    if os["compiler_name"] == "gcc":
+        # Suppress false positive warnings in GCC with stringop-overflow
+        extra_warning_flags += " -Wno-stringop-overflow"
+        # Disable mold, gold and lld linkers.
+        # Use default linker (bfd/ld) which is more lenient with mixed code models
+        cmake_args += " -Duse_mold=OFF -Duse_gold=OFF -Duse_lld=OFF"
+        # Add linker flags for Sanitizers
+        linker_flags += f" -DCMAKE_EXE_LINKER_FLAGS='{linker_relocation_flags} -fsanitize=address,{sanitizers_flags}'"
+        linker_flags += f" -DCMAKE_SHARED_LINKER_FLAGS='{linker_relocation_flags} -fsanitize=address,{sanitizers_flags}'"
+    elif os["compiler_name"] == "clang":
+        # Note: We use $GITHUB_WORKSPACE environment variable which will be expanded by the shell
+        # before CMake processes it. This ensures the compiler receives an absolute path.
+        # CMAKE_SOURCE_DIR won't work here because it's inside CMAKE_CXX_FLAGS string.
+        # GCC doesn't support ignorelist.
+        cxx_flags += " -fsanitize-ignorelist=$GITHUB_WORKSPACE/sanitizers/suppressions/sanitizer-ignorelist.txt"
```

Comment:
```suggestion
        cxx_flags += " -fsanitize-ignorelist=${GITHUB_WORKSPACE}/sanitizers/suppressions/sanitizer-ignorelist.txt"
```

Improves readability and reduces potential for errors.

Reply:
Done